docs/src/domains/steps/non-production-custom-domains.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Set up a custom domain on your non-production environments
+sidebarTitle: (Beta) Non-production custom domains
+weight: 2
+description: Learn how to set up custom domains on your staging and development environments
+betaFlag: true
+---
+
+You can add a custom domain to a staging or development environment without modifying your [routes configuration](../../define-routes/_index.md).
+To do so, you need to attach the new non-production custom domain
+to an existing production custom domain.
+
+{{< note title="Example" >}}
+
+You have two environments, a production environment and a staging environment.
+You've added the `example.com` custom domain to your production environment.
+
+You want to add the `staging.example.com` custom domain to your staging environment.
+To do so, you need to attach the new `staging.example.com` custom domain
+to its corresponding production custom domain `example.com`. 
+
+You can then access your staging environment through `staging.example.com`
+and still access your production environment through `example.com`.
+
+{{< /note >}}
+
+If you have multiple custom domains on your production environment,
+when you create a non-production custom domain,
+you still don't need to update your [routes configuration](../../define-routes/_index.md).
+Platform.sh automatically figures out the routing of your non-production environment
+based on the following elements:
+
+- The production custom domains in your existing [routes configuration](../../define-routes/_index.md)
+- The non-production custom domains attached to each of those production custom domains
+
+## Get started with non-production custom domains
+
+You need:
+
+- A production environment with at least one custom domain already set up
+- At least one non-production (staging or development) environment
```

Comment:
- The environment `Dev` needs to exist for me to run `platform domain:add development.mysite.com --environment Dev --replace mysite.com`?
- Say that environment _did_ exist when I defined this non-production URL, but then the environment was removed. Would re-creating the `Dev` environment after the fact respect the previous custom domain configuration? Or would I need to reconfigure it? A later section implies that the configuration is only removed when 1) the production URL is deleted, or 2) the user explicitly deletes it to define a new one.

Reply:
> The environment Dev needs to exist for me to run platform domain:add development.mysite.com --environment Dev --replace mysite.com?
=> Yes, the environment needs to exist. The CLI returns an error if we try to add a domain to a non-existing environment. I guess that's what we try to say with « _At least one non-production (staging or development) environment_ », but maybe we can be a bit clearer.

> Say that environment did exist when I defined this non-production URL, but then the environment was removed. Would re-creating the Dev environment after the fact respect the previous custom domain configuration? Or would I need to reconfigure it? A later section implies that the configuration is only removed when 1) the production URL is deleted, or 2) the user explicitly deletes it to define a new one.
=> That's a good question, and actually a test that I didn't do. So I tried, and the non-production domain is deleted as well when we delete the environment. If you recreate the environment with the same ID, you have to recreate the non-production domain as well. We should probably add that to the documentation somewhere.

Thanks again for the feedback 🙂